markdown/src/main/java/org/apache/zeppelin/markdown/MarkdownInterpreter.java

```diff
@@ -17,50 +17,83 @@
 
 package org.apache.zeppelin.markdown;
 
-import java.io.IOException;
 import java.util.List;
 import java.util.Properties;
 
 import org.apache.zeppelin.interpreter.Interpreter;
 import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
 import org.apache.zeppelin.interpreter.InterpreterResult;
 import org.apache.zeppelin.interpreter.InterpreterResult.Code;
 import org.apache.zeppelin.interpreter.InterpreterUtils;
 import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
 import org.apache.zeppelin.scheduler.Scheduler;
 import org.apache.zeppelin.scheduler.SchedulerFactory;
-import org.markdown4j.Markdown4jProcessor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * Markdown interpreter for Zeppelin.
- */
-public class Markdown extends Interpreter {
-  private Markdown4jProcessor md;
-  static final Logger LOGGER = LoggerFactory.getLogger(Markdown.class);
+/** MarkdownInterpreter interpreter for Zeppelin. */
+public class MarkdownInterpreter extends Interpreter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MarkdownInterpreter.class);
+
+  private MarkdownParser parser;
+
+  /** Markdown Parser Type. */
+  public enum MarkdownParserType {
+    PEGDOWN {
+      @Override
+      public String toString() {
+        return PARSER_TYPE_PEGDOWN;
+      }
+    },
+
+    MARKDOWN4j {
+      @Override
+      public String toString() {
+        return PARSER_TYPE_MARKDOWN4J;
+      }
+    }
+  }
+
+  public static final String MARKDOWN_PARSER_TYPE = "markdown.parser.type";
+  public static final String PARSER_TYPE_PEGDOWN = "pegdown";
+  public static final String PARSER_TYPE_MARKDOWN4J = "markdown4j";
 
-  public Markdown(Properties property) {
+  public MarkdownInterpreter(Properties property) {
     super(property);
   }
 
+  public static MarkdownParser createMarkdownParser(String parserType) {
+    LOGGER.debug("Creating " + parserType + " markdown interpreter");
+
+    if (MarkdownParserType.PEGDOWN.toString().equals(parserType)) {
+      return new PegdownParser();
+    } else {
+      /** default parser. */
+      return new Markdown4jParser();
+    }
+  }
+
   @Override
   public void open() {
-    md = new Markdown4jProcessor();
+    String parserType = getProperty(MARKDOWN_PARSER_TYPE);
+    parser = createMarkdownParser(parserType);
   }
 
   @Override
   public void close() {}
 
   @Override
-  public InterpreterResult interpret(String st, InterpreterContext interpreterContext) {
+  public InterpreterResult interpret(String markdownText, InterpreterContext interpreterContext) {
     String html;
+
     try {
-      html = md.process(st);
-    } catch (IOException | java.lang.RuntimeException e) {
-      LOGGER.error("Exception in Markdown while interpret ", e);
+      html = parser.render(markdownText);
+    } catch (Exception e) {
```

Comment:
Is there a reason to catch general Exception instead of more specific sub-classes?